python/pyspark/__init__.py

```diff
@@ -36,6 +36,31 @@
       Finer-grained cache persistence levels.
 
 """
+import os
+import sys
+
+import xml.etree.ElementTree as ET
+
+if (os.environ.get("SPARK_HOME", "not found") == "not found"):
```

Comment:
What about starting to add some of the logic from findspark to autodetect SPARK_HOME?